### PR TITLE
Modularize netty

### PR DESCRIPTION
**Pull request by s1monw**
This PR moves all netty related code into `modules/transport-netty` the module is build as a zip file as well as a JAR to serve as a dependency for transport client. For the time being this is required otherwise we have no network based impl. for transport client users. This might be subject to change given that we move forward http client.

### CODE REVIEW

**Comment by jasontedor** [comment]
I'm good with it. Punch it.

File: modules/transport-netty/src/main/java/org/elasticsearch/transport/netty/NettyTransport.java [hunk 1]
```diff
@@ -182,19 +184,25 @@ protected void doStart() {
...
-            clientBootstrap = new ClientBootstrap(
-                new NioClientSocketChannelFactory(
-                    Executors.newCachedThreadPool(daemonThreadFactory(settings, TRANSPORT_CLIENT_BOSS_THREAD_NAME_PREFIX)),
-                    bossCount,
-                    new NioWorkerPool(Executors.newCachedThreadPool(
-                        daemonThreadFactory(settings, TRANSPORT_CLIENT_WORKER_THREAD_NAME_PREFIX)), workerCount),
-                    new HashedWheelTimer(daemonThreadFactory(settings, "transport_client_timer"))));
-        }
+        // this doPrivileged is for SelectorUtil.java that tries to set "sun.nio.ch.bugLevel"
+        AccessController.doPrivileged((PrivilegedAction<Object>) () -> {
```
Comment (rmuir) [inline modules/transport-netty/src/main/java/org/elasticsearch/transport/netty/NettyTransport.java]: I noticed this one in a few places. SelectorUtil does this in a clinit block, so this means as soon as the class gets loaded, its gonna try to do its evil thing.

So an alternative to having this in several places, to consolidate it and reduce the amount of security code, would be to force it to load via `Class.forName` in the `*Plugin` class itself (this is always explicitly initialized by PluginService/unit tests first). Here is an example that does that: https://github.com/elastic/elasticsearch/blob/master/plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsPlugin.java#L39

**Comment by rmuir** [comment]
looks great. I left one suggestion, it should be possible to consolidate the security code to a single place (and it may make it more robust since then its not dependent on class initialization order). We can do it as a followup too.

**Comment by rmuir** [comment]
+1

**Pull request merged**